client/reader/stream/post-lifecycle.jsx

```diff
@@ -13,7 +13,7 @@ import PostPlaceholder from './post-placeholder';
 import PostUnavailable from './post-unavailable';
 import ListGap from 'reader/list-gap';
 import CrossPost from './x-post';
-import { shallowEquals } from 'reader/utils';
+import shallowEquals from '@wordpress/is-shallow-equal';
```

Comment:
A little nitpick: importing as `shallowEquals` is an anomaly ... can we consolidate on `shallowEqual`?

Or even better, use `lib/compare-props` and move the check from `shouldComponentUpdate` to `connect`:
```js
import compareProps from 'lib/compare-props';

connect(
  mapStateToProps,
  null,
  null,
  { areOwnPropsEqual: compareProps( { ignore: [ 'handleClick' ] } ) }
)
```
That also avoids calling `mapStateToProps` on every rerender, not only rerendering the inner component.

The best approach, of course, would be making the incoming `handleClick` prop constant and use vanilla `connect` and `React.Component`.

Reply:
Gutenberg seems to use `isShallowEqual` for the imported name. I'll use that across the codebase unless you prefer otherwise.

Reply:
> Or even better […] The best approach […]

I'd like to land this and we can address that in a dedicated follow-up.

Reply:
Suggested change in #37818